content/en/docs/tasks/administer-cluster/kubelet-config-file.md

```diff
@@ -84,45 +84,309 @@ If `--config` is provided and the values are not specified via the command line,
 
 Main kubelet configuration file contents:
 ```yaml
 apiVersion: kubelet.config.k8s.io/v1beta1
 kind: KubeletConfiguration
 port: 20250
+authorization:
+  mode: Webhook
+  webhook:
+    cacheAuthorizedTTL: "5m"
+    cacheUnauthorizedTTL: "30s"
 serializeImagePulls: false
-evictionHard:
-    memory.available:  "200Mi"
+address: "192.168.0.1"
 ```
 
 Contents of a file in `--config-dir` directory:
 ```yaml
 apiVersion: kubelet.config.k8s.io/v1beta1
 kind: KubeletConfiguration
+authorization:
+  mode: AlwaysAllow
+  webhook:
+    cacheAuthorizedTTL: "8m"
+    cacheUnauthorizedTTL: "45s"
+address: "192.168.0.8"
+```
+
+The resulting configuration will be as follows:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+authorization:
+  mode: AlwaysAllow
+  webhook:
+    cacheAuthorizedTTL: "8m"
+    cacheUnauthorizedTTL: "45s"
 address: "192.168.0.8"
 ```
 
+#### Lists/Slices
+You can overidde the slices/lists values of the kubelet configuration, however, the entire list gets overridden during the merging process. For example, you can override the `clusterDNS` list as follows:
+
+Main kubelet configuration file contents:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+clusterDNS:
+  - "192.168.0.9"
+  - "192.168.0.8"
+```
+
+Contents of a file in `--config-dir` directory:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+clusterDNS:
+  - "192.168.0.2"
+  - "192.168.0.3"
+  - "192.168.0.5"
+```
+
+The resulting configuration will be as follows:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+clusterDNS:
+  - "192.168.0.2"
+  - "192.168.0.3"
+  - "192.168.0.5"
+```
+
+#### Maps, including Nested Structures
+
+Individual fields in maps, regardless of their value types (boolean, string, etc.), can be selectively overridden. However, for `map[string][]string`, the entire list associated with a specific field gets overridden. Let's understand this better with an example, particularly on fields like `featureGates` and `staticPodURLHeader`:
+
+Main kubelet configuration file contents:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+featureGates:
+  MemoryQoS: true
+staticPodURLHeader:
+  kubelet-api-support:
+  - "Authorization: 234APSDFA"
+  - "X-Custom-Header: 123"
+  custom-static-pod:
+  - "Authorization: 223EWRWER"
+  - "X-Custom-Header: 456"
+```
+
+Contents of a file in `--config-dir` directory:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+featureGates:
+  MemoryQoS: false
+  KubeletTracing: true
+staticPodURLHeader:
+  kubelet-api-support:
+  - "Authorization: 8945AFSG1"
```

Comment:
Can you show an example of dropping or not specififying a field in the override file?

Reply:
Fixed